### DISCUSSION TITLE
Disabling normalization of decoding of characters

### DISCUSSION THREAD

**Discussion by SebbeJohansson** [vuejs/router#2307] [Help and Questions] [answered]
Hi! We have issues with the redirect during normalization of encoded characters (| = %7C etc). This is mainly a big issue on Apple devices it seems like.
We would like to disable the feature completely (we are using nuxt) so as to allow the encoded characters to be in its "pure" form.
For example https://domain.com/klader?series=nordwand%7Cgp -> https://domain.com/klader?series=nordwand|gp
Is this possible some how?

**Comment by posva** [comment, marked as answer]
This seems to be [the current behavior](https://play.vuejs.org/#eNqlVFtP2zAU/iteNqlFa+NSEA9Z6GAIjU27IDbtZdlDmpy2gcQ2tlOKSv/7ju3cCoU9TFWV5Fy+71w+e+0Vccb8a+UFXlYILjVZk0RCrOFUCLIhM8kL0luW0ItYFSB5qUFWHp+6z9Zt8mpfLITvciPWoPbxvxcxQvxSQd+lu++Cl0z3e68xrbfnDbwqHWsLNRQix/yJCQwX+5MLyHNuyF6FFD+tWdgHvigtOZtPzkopgVUVExHrRRDSykfWa/LGOvxZmeeX6CSbTZUvJEwaP5pDaiyWgzqSkMXLmu3KtvAlYzdE8+PIo5E3+cjxnVzwAkLa+p/NiKdoeX97HD9Mm+RTY/tHdmDT16430nM4vQG5LUHeB7jLW7QmD2nP9PYycEjrlkKjiW2yXxncEVrFVe6QNkvBXbk97lTSVyi4vL/IlMbHoDI63K7Cho2SGi2Z+VnqWlC1oVJVLTnT0FZcY2nlx5mqlKDIMfltWmnnhiNLOIIxlEvQ0m4G22H1eDuxLbcJ/vOISiJVt9/+2gAu3CiCXfPp71lOVyi+bvBoRAxWttEUZnGZ19g49e48zDFRicyEJgp0KUgeszmqQyvcvFmXc+Irfjw6T+NJjYSnafxkuVvj/D+eBmonkVvosIgFKokzpLIDiyoHMqConTIjr9WMMUfeQmuhAkpLJm7mPu6IthEnh/6BP6IpDrlj9UEVw6nkd8pKN/Ls7A32CQbRFJaa81wNY5E9R/Ek8OTIP/L3aZ5NKaLTjKWwstgGGi+YDbapFWpkls0fNWlkleUgvwudoYa2mo3xurv7bG1altAUmiwgudlhv1YrV/KlBOxuCZ3mdCznoJ37/Mc3WOF74yx4WuYY/YLzChTPS1OjC/tQshTL7sTZaj/ZnWVs/lOdrzQwVTdlCrXTsPF2kWcvtN6We+Afdqao9H0Oyk+UuXPw8hoQczO5vCmXKciAjMWKYLFZSl6PRqN3xlUgXMaGU641LwKyPxIraxdxmmKxjQVZIoawZEJi8hb/FrjKzmGGh78buRh3mVv4toAkSToFBGSEv3GF4G3+Am90jHA=). Note encoding is also handled at the browser level so maybe you are seeing something else changing it

**Reply by SebbeJohansson** [reply]
I'm not sure what you are showing in that example?

My goal is disable the normal behavior where it redirects. Keep in mind that I am in nuxt, but the handling is in the vue router.

**Reply by posva** [reply]
I mean that the `|` is not encoded and that you shouldn't try to control what is encoded or not as it's all part of a spec that needs to be respected for URLs to be valid

**Reply by SebbeJohansson** [reply]
You are misunderstanding the direction. It gets transformed INTO `|` during the routing. Not the other way around. I guess it's decoding from uri into characters.

Look at my example and hopefully it's more clear.

To further show the issue, here it is in a production enviroment: https://www.revolutionrace.com/clothing?series=nordwand%7Cgp

Loading that results in an infinite loop that is based on the decoding and redirect that vue router does.

**Reply by posva** [reply]
I see. Your example confused me because the same URL appears twice like `x -> x` precisely because the `|` gets decoded. If I try to edit the comment I see the difference. I think I've seen that issue reported in Nuxt repo in the past. The encoding of Vue router is intentionally consistent with what the spec expect so the `|` always appearing as `|` is a feature

**Reply by SebbeJohansson** [reply]
Okay good. Sorry that it was confusing.
As I specified, the "redirect" or whatever is technically done causes an infinite redirect in nuxt that they are not planning to change or counter, hence why I'm asking here!

I understand the want for `"pretty"` URLs, but this is causing pages not to work at all (especially on Apple devices). I know this is an issue for many companies, but I understand the reluctance to change it in the application.

Is a setting/workaround completely impossible/out of the question? 😅 For us currently it means that a subset of our links (coming primarily from stuff like newsletters) are completely broken.
![image](https://github.com/user-attachments/assets/35ec10e4-ac78-46e0-beee-cb26cfd53950)

Let me know if I should create another issue on the nuxt GitHub instead.

**Reply by posva** [reply]
I wouldn’t create another issue but instead search for similar ones. I am almost sure the end cause was related till user code and could be fixed
